_resources/port1.0/group/golang-1.0.tcl

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+#
+# This PortGroup accommodates golang projects hosted at GitHub.
+#
+# Usage:
+#
+# PortGroup     golang 1.0
+#
+# go.setup      author project 1.0.0 v # Same as github.setup
+#
+# go.vendors    github.com/dep1/foo abcdef123456... \
+#               github.com/dep2/bar fedcba654321...
+#
+# checksums-append \
+#               dep1-foo-${foo.version}.tar.gz \
+#                   rmd160 abcdef123456... \
+#                   sha256 fedcba654321... \
+#                   size   1234 \
+#               dep2-bar-${bar.version}.tar.gz \
+#                   rmd160 abcdef123456... \
+#                   sha256 fedcba654321... \
+#                   size   4321
+#
+# The go.vendors option expects a list with 2-tuples consisting of package ID
+# and git SHA1.
+#
+# The list of vendors can be found in the Gopkg.lock, glide.lock, etc. file in
+# the upstream source code. The go2port tool (install via MacPorts) can be used
+# to generate a skeleton portfile with precomputed go.vendors and
+# checksums-append values.
+
+PortGroup               github 1.0
+
+options go.bin go.vendors
+
+default go.bin          {${prefix}/bin/go}
+default go.vendors      {}
+
+default platforms       darwin
+
+default use_configure   no
+default dist_subdir     go
+
+default depends_build   port:go
+
+set gopath              ${workpath}/gopath
+default worksrcdir      {${gopath}/src/github.com/${github.author}/${github.project}}
```

Comment:
This assumes that the code came from github, yes?

Reply:
Yes.

Reply:
It might not have. One morning someone is going to put something on gitlab instead. Maybe do this conditionally if the github portgroup is in use?

Reply:
This portgroup as written currently requires the github portgroup (it applies it itself). I did [make reference to this earlier](https://github.com/macports/macports-ports/pull/2550#discussion_r216514776).

Supporting other hosts would require additional special knowledge about e.g. what `master_sites` value to append for each dependency (see `handle_go_vendors`). I thought a GitHub-only implementation would be good enough for a 1.0; if another host becomes popular enough then this portgroup could be expanded or a 1.1 could be added.

Reply:
I think it's ugly only to work with the Github portgroup but I don't mind as such. However, in that case, we need to prominently document at the top that it doesn't work without the github portgroup (and in the docs for the guide), and I would even recommend that the tcl check for the presence of the github portgroup and error out without it. Given that, I'd be okay with it.

Reply:
The golang portgroup was itself applying the github portgroup. But I have now changed things up a bit so this point is moot. I will continue the discussion back in the main thread.